mailpoet/assets/js/src/automation/listing/index.tsx

```diff
@@ -47,14 +48,62 @@ const tableHeaders = [
   { key: 'more' },
 ] as const;
 
-export function AutomationListing({ workflows, loading }: Props): JSX.Element {
+export function AutomationListing({
+  workflows,
+  loading,
+  addNotice,
+}: Props): JSX.Element {
   const history = useHistory();
   const location = useLocation();
   const pageSearch = useMemo(
     () => new URLSearchParams(location.search),
     [location],
   );
 
+  const [allWorkflows, setAllWorkflows] = useState(workflows);
```

Comment:
I'm wondering if this is also a code smell, immediately turning a prop into state.

Reply:
It is suspicious because it's only the state's default value. When the prop changes, it never mirrors that change to the state. You'd need to add `useEffect(() => setAllWorkflows(workflows), [workflows])` as well. Sometimes things like this may be needed, I guess.